Fixes Error - Uncaught Exception Error: write EIO #92

shar-mayank opened the pull request:
- This PR fixes the issue #92. Thanks to @vinaymaurya14, I went through his provided solution and fix the error message that was continuously popping when I execute the `./stealth-run.sh` script. The app won't even quit that time and will through errors whenever any of the shortcut keys were pressed. 
Now both the script and compiled app works as intended. 

Error message was: 
```
A JavaScript error occurred in the main process
Uncaught Exception:
Error: write EIO
at afterWriteDispatched (node:internal/stream_base_commons:160:15)
at writeGeneric (node:internal/stream_base_commons:151:3)
at Socket._writeGeneric (node:net:952:11)
at Socket._write (node:net:964:8)
at writeOrBuffer (node:internal/streams/writable:447:12)
at _write (node:internal/streams/writable:389:10)
at Writable.write (node:internal/streams/writable:393:10)
at console.value (node:internal/console/constructor:304:16) at console.log (node:internal/console/constructor:379:26)
at App.<anonymous> (/Users/xxx/Documents/Projects/interview-coder-withoupaywall-opensource/dist-electron/main.js:515:13)
```
Attached is the video of error:

https://github.com/user-attachments/assets/0b4ed85c-4100-480f-b102-bf57a1e6d68b

- I tested it manually, used every shortcut key commands available and concluded that the app works perfectly! @bhaumikmaan and @anshumansingh01, please look into this.

shar-mayank committed:
Implement SafeLogger for improved logging and replace console logs with safeLogger calls

bhaumikmaan commented on electron/SafeLogger.ts:
```diff
@@ -0,0 +1,111 @@
+        fs.mkdirSync(this.logDir, { recursive: true })
+      }
+    } catch (error) {
+      // Silently fail - we can't log the error since we're creating the logger
```
I don't think we should silently fail this - if this fails there is no way of figuring out what went wrong before or after

bhaumikmaan commented on electron/SafeLogger.ts:
```diff
@@ -0,0 +1,111 @@
+    try {
+      fs.appendFileSync(filePath, message, 'utf8')
+    } catch (error) {
+      // Silently fail - we can't use console.log here as it would cause the same issue
```
This silent failing is fine as everything would be in the log file already.

bhaumikmaan reviewed (changes requested):
Thanks for the PR - it is a good addition. 
We can merge once the review is addressed

shar-mayank commented:
I'll make the necessary changes and will commit soon

bhaumikmaan commented on electron/SafeLogger.ts:
```diff
@@ -0,0 +1,111 @@
+    this.writeToFile(logPath, message)
+  }
+
+  // Clean up old logs (optional - call this periodically)
+  cleanupOldLogs(maxAgeDays: number = 7): void {
```
I don't think it is wise to clear logs at session startup in case one needs to track past errors.
You could do any of the following:
1. Give a button for log clearance
2. Add a message in the initial dialog box with a redirect or a path name so users can do it manually 
3. Clear once a certain size limit is hit like 1 GB threshold (Put a message in the dialog box or the README) that the logs will clear after 1GB

shar-mayank replied:
> I don't think it is wise to clear logs at session startup in case one needs to track past errors.

yeah, you are right!

> You could do any of the following:

> 1. Give a button for log clearance
> 2. Add a message in the initial dialog box with a redirect or a path name so users can do it manually
> 3. Clear once a certain size limit is hit like 1 GB threshold (Put a message in the dialog box or the README) that the logs will clear after 1GB

sure, i'll work on implementing this and will update ASAP!

shar-mayank committed:
feat: Add 1GB auto-cleanup for logs and screenshots with storage info

- Add automatic log directory cleanup when exceeding 1GB in SafeLogger

- Update WelcomeScreen with logs storage location for manual cleanup

- Implement getDirectorySize() and cleanup methods for logs

- Log cleanup actions for debugging purposes

shar-mayank reviewed (commented):
## Add Automatic Storage Management (1GB Cleanup)

### Changes
- **Logs**: Auto-clear when log directory exceeds 1GB
- **UI Updates**: Added storage locations and cleanup info to WelcomeScreen and SettingsDialog

### Implementation
- `SafeLogger.ts`: Added size checking and cleanup before writing logs
- UI components updated to show file locations for manual cleanup

### Testing
Tested manually by different methods like changing the directory size to 1 MB for testing, creating test cases for manual checking and creating a fake log file that exceeds 1GB by `dd if=/dev/zero of=test-large.log bs=1024 count=1126400` and verifying the result manually.

Fixes storage accumulation issues while maintaining app functionality.

@bhaumikmaan, @Ornithopter-pilot, @us1415, @dev-opsss and @Anshumansingh01, please verify and merge. Thank you.

shar-mayank commented:
Note to self: integrate OpenAI's Whisper when free from placements
